shurl-front-end/src/components/App.test.tsx

```diff
@@ -1,9 +1,35 @@
 import React from 'react';
-import { render } from '@testing-library/react';
 import App from './App';
 
-test('renders learn react link', () => {
-  const { getByText } = render(<App />);
-  const linkElement = getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+import { act } from 'react-dom/test-utils';
+import { render, unmountComponentAtNode } from 'react-dom';
+
+let container: HTMLElement;
+
+beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+});
+
+afterEach(() => {
+    unmountComponentAtNode(container);
+    container.remove();
+});
+
+test('can render and create URL', () => {
+    const onClick = jest.fn();
+
+    act(() => {
+        render(<App />, container);
+    });
+
+    const button = container.querySelector('[data-testid=button]');
+    expect(button?.innerHTML).toBe('Create');
+
+    act(() => {
+        button?.dispatchEvent(new MouseEvent('click'));
+    });
+
+    const modal = container.querySelector('[data-testid=modal]');
+    expect(modal).not.toBeInTheDocument();
```

Comment:
This code seems to be a test code that checks for sub-component.
I think it would be better to make a code to check whether the modal is rendered or not.